public/javascripts/SidewalkGallery/src/cards/CardContainer.js

```diff
@@ -0,0 +1,434 @@
+/**
+ * Card Container module. 
+ * This is responsible for managing the Card objects that are to be rendered.
+ * 
+ * @param {*} uiCardContainer UI element tied with this CardContainer.
+ * @returns {CardContainer}
+ * @constructor
+ */
+function CardContainer(uiCardContainer) {
+    let self = this;
+
+    // The number of cards to be shown on a page.
+    const cardsPerPage = 9;
+
+    // Pading between cards.
+    const cardPadding = 25;
+
+    // TODO: Possibly remove if any type of sorting is no longer wanted.
+    let status = {
+        order: 0
+    };
+
+    // Map label type to id.
+    let labelTypeIds = {
+        CurbRamp: 1,
+        NoCurbRamp: 2,
+        Obstacle: 3,
+        SurfaceProblem: 4,
+        Other: 5,
+        Occlusion: 6,
+        NoSidewalk: 7,
+        Assorted: 9
+    };
+
+    // Current label type of cards being shown.
+    let currentLabelType = 'Assorted';
+
+    let currentPage = 1;
+
+    let pageNumberDisplay = null;
+
+    let pagewidth;
+
+    // Map Cards to a CardBucket containing Cards of their label type.
+    let cardsByType = {
+        Assorted: null,
+        CurbRamp: null,
+        NoCurbRamp: null,
+        Obstacle: null,
+        SurfaceProblem: null,
+        Other: null,
+        Occlusion: null,
+        NoSidewalk: null
+    };
+
+    // Keep track of labels we have loaded already as to not grab the same label from the backend.
+    let loadedLabelIds = new Set();
+
+    // Current labels being displayed of current type based off filters.
+    let currentCards = new CardBucket();
+
+    function _init() {
+        pagewidth = uiCardContainer.holder.width();
+
+        // Bind click actions to the forward/backward paging buttons.
+        if (uiCardContainer) {
+            uiCardContainer.nextPage.bind({
+                click: handleNextPageClick
+            })
+            uiCardContainer.prevPage.bind({
+                click: handlePrevPageClick
+            })
+        }
+
+        pageNumberDisplay = document.createElement('h2');
+        pageNumberDisplay.innerText = "1";
+        uiCardContainer.pageNumber.append(pageNumberDisplay);
+        $("#page-control").hide();
+        sg.tagContainer.disable();
+        $("#prev-page").prop("disabled", true);
+        cardsByType[currentLabelType] = new CardBucket();
+
+        // Grab first batch of labels to show.
+        fetchLabelsByType(9, 30, Array.from(loadedLabelIds), function() {
+            render();
+        });
+    }
+
+    function handleNextPageClick() {
+        sg.tracker.push("NextPageClick", null, {
+            From: currentPage,
```

Comment:
No real reason. Would you rather us use all lowercase? like "next_page_click"?

Reply:
Oh "NextPageClick" is like that because that's the format we've been using in our logging on the rest of the website. I was asking about `From: currentPage`. Wasn't sure why "From" was capitalized.

Reply:
ok, I will make the From and To lowercase